Update TimeIntervals to use the new TimeSeriesReferenceVectorData (#482)
NeurodataWithoutBorders/nwb-schema: docs/format/source/format_release_notes.rst

     ``IntracellularRecordingsTable`` together that were recorded simultaneously from different electrodes and for
     storing metdata about simultaneous recordings
   - ``SequentialRecordingsTable`` is a ``DynamicTable`` for grouping different sequential recordings from the
     ``SimultaneousRecordingsTable``  together and storing metadata about sequential recordings
   - ``RepetitionsTable`` a ``DynamicTable`` for grouping different sequential intracellular recordings from the
     ``SequentialRecordingsTable`` together and storing metadata about repetitions
   - ``ExperimentalConditionsTable`` is a ``DynamicTable`` for grouping different intracellular recording repetitions
     from the ``RepetitionsTable`` together and storing metadata about experimental conditions

- Added the new intracellular electrophysiology metadata tables to ``/general/intracellular_ephys`` as part of ``NWBFile`` (#470)

Deprecations
^^^^^^^^^^^^
- ``SweepTable`` has been deprecated in favor of the new intracellular electrophysiology  metadata tables. Use of ``SweepTable``
  is still possible but no longer recommended. (#470)

> - ``/general/intracellular_ephys/filtering`` has been deprated in favor of ``IntracellularElectrode.filtering`` (#470)

Reviewer: ```suggestion
- ``/general/intracellular_ephys/filtering`` has been deprecated in favor of ``IntracellularElectrode.filtering`` (#470)
```

Bug Fixes
^^^^^^^^^

> - Fix incorrect dtype for electrodes table column "filtering" (float -> text) (#478)
> - Remove ``quantity: *`` from the type definitions of ``OptogeneticStimulusSite`` and ``ImagingPlane``.
>   This change improves clarity of the schema to follow best practices. It has no functional effect on the schema. (#472)
> - Require the ``data`` dataset in ``ImageSeries``. Since ``ImageSeries`` is a ``TimeSeries`` and ``TimeSeries.data``
>   is required, ``ImageSeries.data`` should also be a required dataset. Otherwise this creates problems for
>   inheritance and validation. If ``ImageSeries`` data are stored in an external file, then ``ImageSeries.data`` should
>   be set to an empty 3D array. (#481)

Reviewer: ```suggestion
- Fixed incorrect dtype for electrodes table column "filtering" (float -> text) (#478)
- Removed ``quantity: *`` from the type definitions of ``OptogeneticStimulusSite`` and ``ImagingPlane``.
  This change improves clarity of the schema to follow best practices. It has no functional effect on the schema. (#472)
- Updated ``ImageSeries`` to have its ``data`` dataset be required. Since ``ImageSeries`` is a ``TimeSeries`` and ``TimeSeries.data``
  is required, ``ImageSeries.data`` should also be a required dataset. Otherwise this creates problems for
  inheritance and validation. If ``ImageSeries`` data are stored in an external file, then ``ImageSeries.data`` should
  be set to an empty 3D array. (#481)
```
